Automattic/sensei
# Add onboarding navigation

### Changes proposed in this Pull Request

* Add a router by query string to be used in Sensei. For now, it's scoped to the onboarding only, but it's generic, so just moving up the folder, we can use it for other purposes.
* I checked how WooCommerce routes the pages in their wizard and it uses the `react-router` with `@woocommerce/navigation`. For now, I decided don't add these dependencies and create a simple router to work based in query string. I decided it because the code that we would need to write for both cases would be almost the same. And writing a simple implementation, we can evolve how we need and we don't have more dependencies in the project.
* This router basically works using a provider, a route component, and a hook.
  * The provider is the `<QueryStringRouter>` component, and we send as prop only the name of the query string param (`paramName`) that we want to use.
  * To render our routes, we use the `<Route>` component. It receives as prop the route name (`route`) and a prop that says if it's the default route (`defaultRoute`). The default route is used to say the route that should render if our URL doesn't define one. The defaultRoute will **replace** our URL adding the correct query string for this default route.
  * Also, through the `useQueryStringRouter` hook, we can access at any moment the `currentRoute`, and `goTo`. This `goTo` is the way that we'll use to navigate between the routes. In the future if we think is needed a component that has it intrinsic, like the Link from react-router, we can easily create it.

### Testing instructions

* Open page in wp-admin under Sensei LMS -> Onboarding.
* Navigate between the two first steps.
* After go to the second step, click in the first step in the stepper to go to the first again.
* Also, test navigating using the back and next from the browser.

Notice: The navigation (`Navigation` component) must be refactored when data persistence is defined. So we can show the completed steps correctly when the user refreshes the page.
